### PR TITLE
enhancement(sso): sso integration azure adaptation

### PR DESCRIPTION
**Pull request by ungreat**
#### What type of PR is this?

The oauth2 redirectUrl used to concatenate query parameters,  but now uses the state parameter to cache query parameters.

#### Which issue(s) this PR fixes:

Fixes #  https://github.com/oceanbase/odc/issues/3084

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V4324Oauth2UseStateParamDefaultFalse.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import com.oceanbase.odc.common.json.JsonUtils;
+import com.oceanbase.odc.core.migrate.JdbcMigratable;
+import com.oceanbase.odc.core.migrate.Migratable;
+import com.oceanbase.odc.metadb.integration.IntegrationEntity;
+import com.oceanbase.odc.service.integration.model.Oauth2Parameter;
+import com.oceanbase.odc.service.integration.model.SSOIntegrationConfig;
+
+@Migratable(version = "4.3.2.4",
+        description = "Process Historical sso integrations config")
+public class V4324Oauth2UseStateParamDefaultFalse implements JdbcMigratable {
```
Comment (yizhouxw) [inline server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V4324Oauth2UseStateParamDefaultFalse.java:34]: please add unit testing for migrator

File: server/odc-service/src/main/java/com/oceanbase/odc/service/integration/oauth2/Oauth2StateManager.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    private final Cache<String, Map<String, String>> STATE_PARAM_CACHE = CacheBuilder.newBuilder().maximumSize(1000)
+            .expireAfterWrite(10, TimeUnit.MINUTES)
+            .build();
+
+    @Autowired
+    private RequestDispatcher requestDispatcher;
+
+    @SneakyThrows
+    public void addState(String state, String key, String value) {
+        Map<String, String> stringStringMap = STATE_PARAM_CACHE.get(state, HashMap::new);
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/integration/oauth2/Oauth2StateManager.java:53]: addState or setStateParameter?

if state mapping to multiple parameters, use ConcurrentHashMap instead. or line 54 may result into concurrent conflict?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/integration/oauth2/Oauth2StateManager.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        stringStringMap.put(key, value);
+    }
+
+    public void addOdcParam(String state, HttpServletRequest request) {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/integration/oauth2/Oauth2StateManager.java:57]: setOdcParameters

File: server/odc-service/src/main/java/com/oceanbase/odc/service/integration/oauth2/Oauth2StateManager.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        }
+        String testLoginType = request.getParameter(OdcConstants.TEST_LOGIN_TYPE);
+        if (com.oceanbase.odc.common.util.StringUtils.isNotBlank(testLoginType)) {
+            addState(state, OdcConstants.TEST_LOGIN_TYPE, testLoginType);
+        }
+    }
+
+
+    @SneakyThrows
+    public Map<String, String> getStateParameter(String state) {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/integration/oauth2/Oauth2StateManager.java:74]: getStateParameters

File: server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V4324Oauth2UseStateParamDefaultFalse.java (hunk 1 above)
Comment (ungreat) [reply server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V4324Oauth2UseStateParamDefaultFalse.java:34]: done!

File: server/odc-service/src/main/java/com/oceanbase/odc/service/integration/oauth2/Oauth2StateManager.java (hunk 2 above)
Comment (ungreat) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/integration/oauth2/Oauth2StateManager.java:53]: change to setStateParameter.
For the same state, it‘s processed in the same request, which ensures that no other threads compete, so there will be no concurrency issues.

File: server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V4324Oauth2UseStateParamDefaultFalse.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import com.oceanbase.odc.common.json.JsonUtils;
+import com.oceanbase.odc.core.migrate.JdbcMigratable;
+import com.oceanbase.odc.core.migrate.Migratable;
+import com.oceanbase.odc.metadb.integration.IntegrationEntity;
+import com.oceanbase.odc.service.integration.model.Oauth2Parameter;
+import com.oceanbase.odc.service.integration.model.SSOIntegrationConfig;
+
+@Migratable(version = "4.3.2.4",
```
Comment (yhilmare) [inline server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V4324Oauth2UseStateParamDefaultFalse.java:32]: you may use 4.3.2.5, 4.3.2.4 is occupied

Comment (ungreat) [reply server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V4324Oauth2UseStateParamDefaultFalse.java]: done!

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**